ReactAndroid/src/main/java/com/facebook/react/shell/MainPackageConfig.java

```diff
@@ -10,31 +10,52 @@
 package com.facebook.react.shell;
 
 import com.facebook.imagepipeline.core.ImagePipelineConfig;
+import com.facebook.react.modules.network.DefaultOkHttpProvider;
+import com.facebook.react.modules.network.OkHttpClientProvider;
+
+import javax.annotation.Nullable;
 
 /**
  * Configuration for {@link MainReactPackage}
  */
 public class MainPackageConfig {
 
-  private ImagePipelineConfig mFrescoConfig;
+  private final @Nullable OkHttpClientProvider mHttpClientProvider;
+
+  private final ImagePipelineConfig mFrescoConfig;
 
   private MainPackageConfig(Builder builder) {
     mFrescoConfig = builder.mFrescoConfig;
+    mHttpClientProvider = builder.mHttpClientProvider;
   }
 
   public ImagePipelineConfig getFrescoConfig() {
     return mFrescoConfig;
   }
 
+  public OkHttpClientProvider getHttpClientProvider() {
+    return mHttpClientProvider != null ? mHttpClientProvider : new DefaultOkHttpProvider();
+  }
+
   public static class Builder {
 
     private ImagePipelineConfig mFrescoConfig;
+    private OkHttpClientProvider mHttpClientProvider;
 
     public Builder setFrescoConfig(ImagePipelineConfig frescoConfig) {
       mFrescoConfig = frescoConfig;
       return this;
     }
 
+    /**
+     * Option to inject configured OkHttpClient to react-native networking subsystem.
+     * This will be used only by the fetch networking calls (Fresco library, for eg, will not use this client).
```

Comment:
nit: Wording:

Allows to provide a custom HTTP client provider that will be used for networking calls from JavaScript (especially the fetch API). This can be useful for example in hybrid apps which have their own configuration of the HTTP client, such as custom cookies etc.

This does not configure the networking client the Image JavaScript component uses when fetching images.